src/controller/c/motor.c

```diff
@@ -174,6 +176,8 @@ static void motor_read_answer(WbDevice *d, WbRequest *r) {
       m->control_i = m->previous_control_i;
       m->control_d = m->previous_control_d;
       m->position = request_read_double(r);
+      m->velocity = request_read_double(r);
```

Comment:
Why did you add velocity here? It doesn't seem to be useful to anything.

Reply:
Because when the controller sets the velocity, webots is informed about it and in turn it relays the command to its siblings. If these siblings don't notify libController about the change in velocity, any call to `wb_motor_get_velocity` that directly targets one of these siblings would return their old value, not the new one.